src/validation-dispatcher/record-dispatcher.ts

```diff
@@ -1,10 +1,38 @@
-import { ValidationResult, RecordValidationSchema } from '../model';
+import {
+  ValidationResult,
+  RecordValidationSchema,
+  RecordValidationFunctionAsync,
+} from '../model';
+import { isFunction } from '../helper';
+import { convertRecordValidationToAsyncIfNeeded } from '../mapper';
+
+const getValidationFn = (
+  recordValidation: RecordValidationSchema
+): RecordValidationFunctionAsync => {
+  if (!recordValidation) return null;
+
+  let recordValidationFunction: RecordValidationFunctionAsync = null;
+
+  if (isFunction(recordValidation)) {
+    recordValidationFunction = convertRecordValidationToAsyncIfNeeded(
```

Comment:
Why conver it again? We had convert it in `validationEngine`, isn't it?

Reply:
That's a good point, we have to balance, one one hand we would created a StrictRecordValidationSchema that contains always the async function, it's a bit more of overhead on typings but we avoid have to make this security check twice.

On the other we can keep the same type and pay this overhead, I think is best to go for the first option what do you think?

Reply:
I agree